plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/part/AssociationPart.java

```diff
@@ -103,10 +105,11 @@ protected void createEditPolicies() {
 
     @Override
     protected IFigure createFigure() {
-        PolylineConnection conn = new PolylineConnection();
-
+        final DBPPreferenceStore store = ERDUIActivator.getDefault().getPreferences();
+        ERDConnectionRouterDescriptor connectionRouterDescriptor = ERDConnectionRouterRegistry.getInstance()
```

Comment:
You use `store.getString(ERDUIConstants.PREF_ROUTING_TYPE)` in many places. Let's move this logic into registry function. Also please cache it (preference reading is relatively slow and you call this call very frequently).